bots/teams_bot_adapter/teams_chromedriver_payload.js

```diff
@@ -1500,27 +1500,35 @@ const processClosedCaptionData = (item) => {
     window.ws.sendClosedCaptionUpdate(itemConverted);
 }
 
-const handleMainChannelEvent = (event) => {
-    //realConsole?.log('handleMainChannelEvent', event);
-    const decodedData = new Uint8Array(event.data);
-
-    const jsonRawString = new TextDecoder().decode(decodedData);
-    //realConsole?.log('handleMainChannelEvent jsonRawString', jsonRawString);
-    
-    // Find the start of the JSON data (looking for '[' or '{' character)
-    let jsonStart = 0;
+const decodeMainChannelData = (data) => {
+    const decodedData = new Uint8Array(data);
     for (let i = 0; i < decodedData.length; i++) {
         if (decodedData[i] === 91 || decodedData[i] === 123) { // ASCII code for '[' or '{'
-            jsonStart = i;
-            break;
+            const candidateJsonString = new TextDecoder().decode(decodedData.slice(i));
+            try {
+                return JSON.parse(candidateJsonString);
+            }
+            catch(e) {
+                if (e instanceof SyntaxError) {
+                    // If JSON parsing fails, continue looking for the next '[' or '{' character
+                    // as binary data may contain bytes that coincidentally match these character codes
+                    continue;
+                }
+                realConsole?.error('Failed to parse main channel data:', e);
+                return;            
+            }        
         }
     }
-    
-    // Extract and parse the JSON portion
-    const jsonString = new TextDecoder().decode(decodedData.slice(jsonStart));
+}
+
+const handleMainChannelEvent = (event) => {
     try {
-        const parsedData = JSON.parse(jsonString);
-        //realConsole?.log('handleMainChannelEvent parsedData', parsedData);
+        const parsedData = decodeMainChannelData(event.data);
+        if (!parsedData) {
+            realConsole?.error('handleMainChannelEvent: Failed to parse main channel data, returning, data:', data);
```

Comment:
`data` appears to be undefined in this scope. Did you mean to use `event.data` instead?
```suggestion
            realConsole?.error('handleMainChannelEvent: Failed to parse main channel data, returning, data:', event.data);
```

Reply:
Ooh yea thanks for catching that! Applied your suggestion